tests/uptane_vector_tests.cc

```diff
@@ -15,178 +15,152 @@
-          }
-        }
-        for (const auto& t : vector["image_repo"]["targets"]) {
-          if (!t["is_success"].asBool()) {
-            should_fail = true;
-            break;
-          }
+    bool should_fail = false;
+    if (!vector["director"]["update"]["is_success"].asBool() ||
+        !vector["image_repo"]["update"]["is_success"].asBool()) {
+      should_fail = true;
+    } else {
+      for (const auto& t : vector["director"]["targets"]) {
+        if (!t["is_success"].asBool()) {
+          should_fail = true;
+          break;
         }
       }
-
-      try {
-        /* Fetch metadata from the director.
-         * Check metadata from the director.
-         * Identify targets for known ECUs.
-         * Fetch metadata from the images repo.
-         * Check metadata from the images repo. */
-        if (!uptane_client->uptaneIteration()) {
-          if (should_fail) {
-            throw uptane_client->getLastException();
-          }
-          std::cout << "uptaneIteration unexpectedly failed.\n";
-          return false;
-        }
-        std::vector<Uptane::Target> updates;
-        if (!uptane_client->uptaneOfflineIteration(&updates, nullptr)) {
-          std::cout << "uptaneOfflineIteration unexpectedly failed.\n";
-          return false;
-        }
-        if (updates.size()) {
-          /* Download a binary package.
-           * Verify a binary package. */
-          result::Download result = uptane_client->downloadImages(updates);
-          if (result.status != result::DownloadStatus::kSuccess) {
-            if (should_fail) {
-              throw uptane_client->getLastException();
-            }
-            std::cout << "downloadImages unexpectedly failed.\n";
-            return false;
-          }
+      for (const auto& t : vector["image_repo"]["targets"]) {
+        if (!t["is_success"].asBool()) {
+          should_fail = true;
+          break;
         }
+      }
+    }
 
-      } catch (const Uptane::Exception& e) {
-        if (match_error(vector, e)) {
-          continue;
+    try {
+      /* Fetch metadata from the director.
+       * Check metadata from the director.
+       * Identify targets for known ECUs.
+       * Fetch metadata from the images repo.
+       * Check metadata from the images repo. */
+      if (!uptane_client->uptaneIteration()) {
+        ASSERT_TRUE(should_fail) << "uptaneIteration unexpectedly failed.";
+        throw uptane_client->getLastException();
+      }
+      std::vector<Uptane::Target> updates;
+      ASSERT_TRUE(uptane_client->uptaneOfflineIteration(&updates, nullptr))
+          << "uptaneOfflineIteration unexpectedly failed.";
+      if (updates.size()) {
+        /* Download a binary package.
+         * Verify a binary package. */
+        result::Download result = uptane_client->downloadImages(updates);
+        if (result.status != result::DownloadStatus::kSuccess) {
+          ASSERT_TRUE(should_fail) << "downloadImages unexpectedly failed.";
```

Comment:
What about a case when should_fail is true and result.status is kSuccess ? Should such case ne handled too ?